tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/S3SingleSaveBlobStoreDAO.java

```diff
@@ -0,0 +1,37 @@
+package com.linagora.tmail.blob.blobid.list;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.blob.api.ObjectStoreException;
+import org.apache.james.blob.objectstorage.aws.S3BlobStoreConfiguration;
+import org.apache.james.blob.objectstorage.aws.S3BlobStoreDAO;
+import reactor.core.publisher.Mono;
+
+import javax.inject.Inject;
+import java.util.Objects;
+
+public class S3SingleSaveBlobStoreDAO extends S3BlobStoreDAO {
```

Comment:
Favor composition over inheritance.

-> Have a `private final BlobStoreDAO underlying;` field and only **implement** things (BlobStoreDAO) not extend.

Genericity: only assume you have a BlobStoreDAO then you don't need the blob-memory and blob-cassandra dependencies.

Reply:
I am still confusing. If `S3SingleSaveBlobStoreDAO implement BlobStoreDAO`, I need to override all method?
Will I clone all code from `S3BlobStoreDAO` to here?

Reply:
No, you need to inject the BlobStoreDAO in your code, and only implement the extra methods you need, as @chibenwa said. Guice will take care of injecting the right DAO (memory or Cassandra)